src/frontend/src/binder/expr/function.rs

```diff
@@ -241,10 +241,24 @@ impl Binder {
                 if let Ok(expr) = UdfContext::extract_udf_expression(ast) {
                     self.set_udf_binding_flag();
                     let bind_result = self.bind_expr(expr);
-                    self.unset_udf_binding_flag();
+
+                    // We should properly decrement global count after a successful binding
+                    // Since the subsequent `unset flag` operation relies on global counting
+                    self.udf_context.decr_global_count();
+
+                    // Only the top-most sql udf binding should unset the flag
+                    // Otherwise the subsequent binding may not be able to
+                    // find the corresponding context, consider the following example:
+                    // e.g., `select add_wrapper(a INT, b INT) returns int language sql as 'select add(a, b) + a';`
+                    // The inner `add` should not unset the flag, otherwise the `a` will be treated as
+                    // a normal column, which is then invalid in this context.
+                    if self.udf_context.global_count() == 0 {
+                        self.unset_udf_binding_flag();
+                    }
```

Comment:
Looks like `udf_binding_flag` can be inferred by `udf_global_counter != 0`. Can we remove `udf_binding_flag` now?

Reply:
I think so, let me do a tiny refactor.